### DISCUSSION TITLE
Using useSWR() without key Doesn't return any data

### DISCUSSION THREAD

**Discussion by mohanas-scg** [vercel/swr#2808] [Q&A] [answered]
```ts
export async function getData() {
  let res = await fetch("/api/product/", { cache: "no-store" });
  if (!res.ok) {
    throw new Error("Failed to fetch data");
  }
  return res.json();
}
```

```ts
const {
    data: productData,
    isLoading,
    mutate,
  }: SWRResponse<resProductFields[], any, any> = useSWR<resProductFields[]>(getData);
```

Here without key useSWR doesn't return any data can anyone explain this scenario?

**Comment by koba04** [comment]
Yeah, the key param is required, so you need to put a key like `/api/product/` as the key.
https://swr.vercel.app/docs/api#parameters

**Reply by mohanas-scg** [reply]
Yes I agree, it is working for me but, why can't we just put the URL directly inside the fetcher function instead of passing it as a key? (is that key anything to do with caching👀?)

**Reply by koba04** [reply, marked as answer]
Yes, that is correct; we use the key internally to store data like cache and some internal state.
You can also pass a non-URL string as the key.

```jsx
useSWR("product", getData);
```

**Reply by mohanas-scg** [reply]
Thank you @koba04 for the answer.
